.github/workflows/test.yml

```diff
@@ -232,3 +232,44 @@ jobs:
 
       - name: Run Forge example tests using remote network with a pinned block number
         run: forge test --remappings hedera-forking=. -vv --fork-url https://mainnet.hashio.io/api --fork-block-number 72433403
+
+  hardhat-example-tests:
+    name: 'Hardhat example tests w/mainnet'
+    runs-on: smart-contracts-linux-medium
+    steps:
+      - name: Harden Runner
+        uses: step-security/harden-runner@5c7944e73c4c2a096b17a9cb74d65b6c2bbafbde # v2.9.1
+        with:
+          egress-policy: audit
+
+      - name: Checkout Code
+        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+        with:
+          submodules: recursive
+
+      - name: Set up Node.js
+        uses: actions/setup-node@1e60f620b9541d16bece96c5465dc8ee9832be0b # v4.0.3
+        with:
+          node-version: '20'
+
+      - name: Run Forge example tests using remote network with a pinned block number
+        run: cd examples/hardhat-hts && npm install
+
+      - name: Install Foundry
+        uses: step-security/foundry-toolchain@01e2d127ea061e2ce8fa34340de71ca8518f093e # v1.2.1
+        with:
+          version: nightly
+
+      - name: Run Forge build
+        run: forge build
+
+      - name: Making sure the contracts are up to date
+        run: |
+          rm -rf examples/hardhat-hts/node_modules/@hashgraph/system-contracts-forking/contracts
+          mv contracts examples/hardhat-hts/node_modules/@hashgraph/system-contracts-forking/contracts
+
+      - name: Moving build output to the plugin directory, where it's missing
+        run: mv out examples/hardhat-hts/node_modules/@hashgraph/system-contracts-forking
+
+      - name: Run Example tests
+        run: cd examples/hardhat-hts && npx hardhat test --grep USDC && npx hardhat test --grep NFT
```

Comment:
there is an npm script to run hardhat tests, `npm run test`, let's use that one instead